pkg/interface/chat/src/js/components/lib/ship-search.js

```diff
@@ -0,0 +1,339 @@
+      return null;
+    }
+
+    const popoverClasses = (popover && " absolute ") || " ";
+    return (
+      <div
+        style={
+          popover
+            ? {
+                bottom: "90%",
+                left: "48px"
+              }
+            : {}
+        }
+        className={
+          "black white-d bg-white bg-gray0-d " +
+            "w7 pv3 z-1 mt1 ba b--gray1-d b--gray4" +
+            popoverClasses +
+            className || ""
+        }
+      >
+        {suggestions.slice(0, 5).map(ship => (
+          <ShipSearchItem
+            onSelect={onSelect}
+            key={ship}
+            selected={selected}
+            contacts={contacts}
+            ship={ship}
+          />
+        ))}
+      </div>
+    );
+  }
+}
+
+export class ShipSearchInput extends Component {
+  constructor() {
+    super();
+    this.state = {
+      searchTerm: ""
+    };
+
+    this.inputRef = React.createRef();
+    this.popoverRef = React.createRef();
+
+    this.search = this.search.bind(this);
+
+    this.onClick = this.onClick.bind(this);
+  }
+
+  onClick(event) {
+    const { popoverRef } = this;
+    // Do nothing if clicking ref's element or descendent elements
+    if (!popoverRef.current || popoverRef.current.contains(event.target)) {
+      return;
+    }
+
+    this.props.onDismiss();
+  }
+
+  componentDidMount() {
+    document.addEventListener("mousedown", this.onClick);
+    document.addEventListener("touchstart", this.onClick);
+    this.inputRef.current.focus();
+  }
+
+  componentWillUnmount() {
+    document.removeEventListener("mousedown", this.onClick);
+    document.removeEventListener("touchstart", this.onClick);
+  }
+
+  search(e) {
+    const searchTerm = e.target.value;
+    this.setState({ searchTerm });
+  }
+
+  render() {
+    const { state, props } = this;
+
+    return (
+      <div
+        ref={this.popoverRef}
+        style={{ top: "150%", left: "-80px" }}
+        className="b--gray2 b--solid ba absolute bg-white bg-gray0-d shadow-5"
+      >
+        <textarea
+          style={{ resize: "none" }}
+          className="ma2 pa2 b--gray4 ba b--solid w7 db bg-gray0-d white-d"
+          rows={1}
+          autocapitalise="none"
+          autoFocus={
+            /Android|webOS|iPhone|iPad|iPod|BlackBerry/i.test(
+              navigator.userAgent
+            )
+              ? false
+              : true
+          }
+          placeholder="Search for a ship"
+          value={state.searchTerm}
+          onChange={this.search}
+          ref={this.inputRef}
+        />
+        <ShipSearch
+          contacts={props.contacts}
+          candidates={props.candidates}
+          searchTerm={deSig(state.searchTerm)}
+          inputRef={this.inputRef.current}
+          onSelect={props.onSelect}
+          onDismiss={props.onDismiss}
+          suggestEmpty
+        />
+      </div>
+    );
+  }
+}
+
+// import { useEffect } from 'react';
+
+// // via https:// usehooks.com/useOnClickOutside/
```

Comment:
We will *never* incorporate hooks into any of our apps, so please remove this commented out code, as it's not a viable route that we would implement.